### ISSUE TITLE
Help with running my own compilation

### ISSUE THREAD

**Issue by nukee86** [otya128/winevdm#1427]
I have VS2017 installed, with msys2 and mingw installed with this command:

`pacman -S --needed base-devel mingw-w64-i686-toolchain`

I changed path in PropertySheet.props to: `c:\msys64\usr\bin\`. I cleaned and build. There were alot of warnings and when i run emulator with my prog I got: `ABNORMAL PROGRAM TERMINATION`. I guess there shouldn't be so much warnings during compilation. Can someone hint me how to succesfully compile it? Maybe I should try different version of as, bison and flex(don't know which one)? I'm working on Windows 10 64bit.

**Comment by cracyc** [comment]
Which are you trying to build with, GCC or MSVC?

**Comment by nukee86** [comment]
MSVC. Here are as, bison and flex I use:

```
c:\msys64\usr\bin>as --version
GNU assembler (GNU Binutils) 2.42
Copyright (C) 2024 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or later.
This program has absolutely no warranty.
This assembler was configured for a target of 'x86_64-pc-msys'.

c:\msys64\usr\bin>bison --version
bison (GNU Bison) 3.8.2
Written by Robert Corbett and Richard Stallman.

Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

c:\msys64\usr\bin>flex --version
flex 2.6.4
```

List of my VS2017 install packages:
![vspkg](https://github.com/user-attachments/assets/322e6c88-dc84-42f2-bbc1-e41162fda22f)

**Comment by cracyc** [comment]
There are lots of warnings, that's expected.  The debugger ought to give a hint as why it's failing.

**Comment by nukee86** [comment]
I tested building from command line as in AppVeyor. Tried the route from readme. Still can't make it to run as stable as in newest release from AppVeyor. I can run it from command line(either otvdm or otvdmw). Everything works fine, until I try to make conection to device, then I get following error:
![newconn](https://github.com/user-attachments/assets/bf504bba-af39-425f-a623-607445052d6a)

Raw terminal connection is working without the error.

If I try to open otvdmw and select file in prompt, I got following error:
![openprompt](https://github.com/user-attachments/assets/ca8a8538-f21e-4eb5-9e56-af88d2b79444)

Could you please look at my output logs and tell if it looks how it supposed to?
[output.txt](https://github.com/user-attachments/files/16261183/output.txt)

**Comment by nukee86** [comment]
Yeah, I guess so. I set a fresh VM with Windows 10 64 bit , installed fresh VS2017, still got the same result as above. May I know what operating system do you use to compile the project? Would it be possible to get your output file? I'd like to check which warnings are irrelevant.

**Comment by cracyc** [comment]
Windows 11 but I did it on Windows 10 before.  I'd suggest downloading a fresh tree and just building from msbuild rather then loading into visual studio.

**Comment by nukee86** [comment]
I tried with fresh VS2017, 2019 and 2022 installs. no success - "Abnormal program termination" error. I almost give up, but tried to run latest release. It turned out it was a problem with my virtual machine. I created a new one, with VS2022 compiled everything as above and my program run just as latest build. I'm only worried about those warnings - despite fresh install and building from msbuild it still go these warnings. Sadly I can't see specific warnings from appveyor and can't compare them to mine. In the meantime I'll try to see what caused the error in my previous VM - seems like some installed program messed up. Thank you for your help.
